index.d.ts

```diff
@@ -0,0 +1,103 @@
+/// <reference types="node" />
+
+declare module 'cloud-config-client' {
+
+    import http = require('http');
+    import https = require('https');
+
+    export function load(options: Options, callback?: LoadCallback): Promise<Config|Error>;
+
+    export abstract class Config {
+        constructor(data: ConfigData, context: { [key: string]: any });
+
+        properties(): { [key: string]: any }
+
+        raw(): { [key: string]: any }
+
+        get(keyParts: string): any
+
+        forEach(callback: (property: string, value: string) => void, includeOverridden: boolean): void
+
+        toObject(): { [key: string]: any }
+
+        toString(spaces: number): string
+    }
+
+    interface ConfigFile {
+        /** file name */
+        name: string
+
+        /** configuration properties */
+        source: ConfigSource
+    }
+
+    interface ConfigSource {}
```

Comment:
Let's use a type instead of an interface here 

```
type ConfigSource {
  [key: string]: any
}
```

Reply:
Yes sure, you can adjust as you think it is the best.